packages/picasso.js/src/core/component/component-factory.js

```diff
@@ -351,19 +351,32 @@ function componentFactory(definition, context = {}) {
         createDockDefinition(settings, preferredSize, chart.logger()),
         dockConfigCallbackContext
       );
+      brushArgs.config = settings.brush || {};
     }
 
     if (settings.scale) {
       scale = chart.scale(settings.scale);
     }
 
     if (settings.data) {
-      data = extractData(
+      const { rendererSettings } = settings;
+      const progressive = typeof rendererSettings?.progressive === 'function' && rendererSettings.progressive();
+      const extracted = extractData(
         settings.data,
         { dataset: chart.dataset, collection: chart.dataCollection },
         { logger: chart.logger() },
         chart.dataCollection
       );
+      if (!progressive) {
+        data = extracted;
+      } else if (progressive.isFirst) {
+        data = extracted;
+        if (data.items) {
```

Comment:
why ` if (data.items) {` should it not always update data.items

Reply:
but don't you need to set data.items to an empty array in the case?
or how does the data.items.push(...) on the next progressive render without that.